lib/internal/Magento/Framework/ObjectManager/Code/Generator/Repository.php

```diff
@@ -228,7 +228,6 @@ protected function _getGetMethod()
             'parameters' => [
                 [
                     'name' => 'id',
-                    'type' => 'int',
```

Comment:
> I think that should be tackled with a further PR - do you agree?

Do you mean in the separated PR?
I think it should be done in the scope of one PR and covered with additional tests because existing tests are not enough.